File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        int n = nums.length;
        int sum=0;
        int count=0;
        HashMap<Integer, Integer> map = new HashMap<>();
        map.put(0, 1);
      
        for(int v : nums){
            sum+=v;
            int temp = sum-k;
            if(map.containsKey(temp)){
                count+=map.get(temp);
            }
            map.put(sum, map.getOrDefault(sum, 0)+1);
        }
    return count;
        
    }
}

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution(object):
    def subarraySum(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        freq = {0:1}
        sum = 0
        count = 0
        for num in nums:
            sum+=num
            complement = sum-k
            if complement in freq:
                count+=freq[complement]
            freq[sum] = freq.get(sum, 0)+1
        return count

        
        